src/main/java/in/org/projecteka/hiu/ClientError.java

```diff
@@ -94,12 +89,12 @@ public static ClientError invalidDataFromGateway() {
                 new ErrorRepresentation(new Error(ErrorCode.INVALID_DATA_FROM_GATEWAY, "Invalid Data from Gateway. Must have either a payload or error")));
     }
 
-    public static ClientError invalidDataFromGateway(String errMsg) {
-        return new ClientError(BAD_REQUEST,
-                new ErrorRepresentation(new Error(ErrorCode.INVALID_DATA_FROM_GATEWAY, errMsg)));
-    }
-
     public static ClientError gatewayTimeOut() {
         return new ClientError(GATEWAY_TIMEOUT, new ErrorRepresentation(new Error(NO_RESULT_FROM_GATEWAY, "Didn't receive any result from Gateway")));
```

Comment:
I think it should be simple GATEWAY_TIMEOUT error, with message "Could not connect to Gateway". You have not received the results yet.

Reply:
"Could not connect to Gateway" may also tell that Gateway service is not up?

Reply:
I think a generic msg is fine. If not, can you think of an appropriate one - becacuse its not about receiving result from Gateway - right?

Reply:
Yes. This is shown after the response time has passed.

Reply:
Done